controllers/helpers.go

```diff
@@ -258,6 +263,16 @@ func userAssignedIdentityCloudProviderConfig(d azure.ClusterScoper, identityID s
 	return controlPlaneConfig, workerConfig
 }
 
+func workloadIdentityCloudProviderConfig(d azure.ClusterScoper) (cpConfig *CloudProviderConfig, wkConfig *CloudProviderConfig) {
+	controlPlaneConfig, workerConfig := newCloudProviderConfig(d)
+	// secret is not needed in workload identity.
+	controlPlaneConfig.AadClientSecret = ""
+	controlPlaneConfig.UseFederatedWorkloadIdentityExtension = true
+	workerConfig.AadClientSecret = ""
+	controlPlaneConfig.AADFederatedTokenFile = scope.AzureTokenFilePath
```

Comment:
how does this file get on the VM? don't we need to enable workload identity for the workload cluster in order to be able to use for cloud-provider-azure?